### PR TITLE
Fix #368: Highlight opened file on load

### LINKED ISSUE #368: Highlight opened file on load
On load, brackets does not highlight the currently open file in the filetree. On load, this file should be in a selected state.

### PR DESCRIPTION
**Pull request by Th30**
Hey,

removed some code that was returning null points, this fixes the highlight the proper file on open. There might be more to remove.

The problem was that it still contained code from the brackets application (main adobe repo), that was returning null because it doesn't exist thimble.
As seen below in brackets:

screen shot 2017-04-04 at 3 30 05 pm

The problem was here:
image

In the image, I did a quick fix, but it merely bypassed the problem not really solving it. 

For the PR I made a more thorough fix, so hopefully it's merge ready, but let me know if there's something i missed or need add.

I've modified the comments as well to reflect what's happening now.

Edit:
Fixes #368

### CODE REVIEW

**Comment by flukeout** [comment]
Hey, cool - will test it out now!

**Comment by flukeout** [comment]
@Th30 Hey awesome, in testing this - it worked as expected. The file that is open in the editor is also highlighted in the file-tree! 👍 

Could you please add ``Fixes #368`` to your PR description so that we close the associated issue automatically when we merge this PR?

Are you planning to work on it some more, or is this ready for a code review?

**Comment by Th30** [comment]
@flukeout Hey I did as requested:). I'll gladly continue to structure the code if required, but as far as I can see it's code review ready.

**Comment by flukeout** [comment]
Okay great - @humphd - would you mind reviewing this code? Tested that it works.

File: src/project/FileViewController.js [hunk 1]
```diff
@@ -61,7 +61,6 @@ define(function (require, exports, module) {
      * @private
      */
     var _curDocChangedDueToMe = false;
-    var WORKING_SET_VIEW = "WorkingSetView";
```
Comment (humphd) [inline src/project/FileViewController.js]: We can't do this.  You're assuming that this is only used internally in this file, but if you do a `git grep WORKING_SET_VIEW` you'll see that it's used in a lot of places.  This is a public API, and you have to maintain it for other call sites to not break.  When you make changes like this, *always* look at who else in the code uses it.  Don't assume that things are isolated to the particular case you're solving.

I've filed https://github.com/mozilla/brackets/issues/692 to deal with removing this code properly.

File: src/project/FileViewController.js [hunk 2]
```diff
@@ -70,14 +69,6 @@ define(function (require, exports, module) {
      */
     var _fileSelectionFocus = PROJECT_MANAGER;
 
-    // Due to circular dependencies, not safe to call on() directly
-    /**
-     * Change the doc selection to the working set when ever a new file is added to the working set
-     */
-    EventDispatcher.on_duringInit(MainViewManager, "workingSetAdd", function (event, addedFile) {
-        _fileSelectionFocus = WORKING_SET_VIEW;
```
Comment (humphd) [inline src/project/FileViewController.js]: I think you should just change this line to the following (NOTE: the comment too):

```js
// XXXBramble: always keep focus on the Project Manager vs. Working Set View
_fileSelectionFocus = PROJECT_MANAGER;
```

If that works OK, I think it's better to keep the events in place.  If that doesn't work, we can explore removing this.

File: src/project/FileViewController.js [hunk 3]
```diff
@@ -87,11 +78,7 @@ define(function (require, exports, module) {
         if (!_curDocChangedDueToMe) {
             // The the cause of the doc change was not openAndSelectDocument, so pick the best fileSelectionFocus
             perfTimerName = PerfUtils.markStart("FileViewController._oncurrentFileChange():\t" + (file ? (file.fullPath) : "(no open file)"));
-            if (file && MainViewManager.findInWorkingSet(paneId,  file.fullPath) !== -1) {
-                _fileSelectionFocus = WORKING_SET_VIEW;
-            } else {
-                _fileSelectionFocus = PROJECT_MANAGER;
-            }
```
Comment (humphd) [inline src/project/FileViewController.js]: Add a comment above this line:

```
// XXXBramble: always keep focus on the Project Manager vs. Working Set View
```

File: src/project/FileViewController.js [hunk 4]
```diff
@@ -120,14 +107,9 @@ define(function (require, exports, module) {
...
      * Modifies the selection focus in the project side bar. A file can either be selected
      * in the working set (the open files) or in the file tree, but not both.
-     * @param {String} fileSelectionFocus - either PROJECT_MANAGER or WORKING_SET_VIEW
+     * @param {String} fileSelectionFocus - PROJECT_MANAGER
      */
     function setFileViewFocus(fileSelectionFocus) {
-        if (fileSelectionFocus !== PROJECT_MANAGER && fileSelectionFocus !== WORKING_SET_VIEW) {
-            console.error("Bad parameter passed to FileViewController.setFileViewFocus");
-            return;
-        }
```
Comment (humphd) [inline src/project/FileViewController.js]: Keep this `if` check, since we don't want to break assumptions elsewhere in the code.  However, after it do this:

```js
// XXXBramble: always keep focus on the Project Manager vs. Working Set View
fileSelectionFocus = PROJECT_MANAGER;
```

File: src/project/FileViewController.js [hunk 5]
```diff
@@ -159,11 +141,6 @@ define(function (require, exports, module) {
             return window.event && (_secondPaneContext() || _firstPaneContext());
         }
 
-        if (fileSelectionFocus !== PROJECT_MANAGER && fileSelectionFocus !== WORKING_SET_VIEW) {
-            console.error("Bad parameter passed to FileViewController.openAndSelectDocument");
-            return;
-        }
```
Comment (humphd) [inline src/project/FileViewController.js]: Leave this in for the same reason stated above.

File: src/project/FileViewController.js [hunk 6]
```diff
@@ -210,12 +187,11 @@ define(function (require, exports, module) {
...
         // documentSelectionFocusChange so the views change their selection
         promise.done(function (file) {
-            // CMD_ADD_TO_WORKINGSET_AND_OPEN command sets the current document. Update the
-            // selection focus only if doc is not null. When double-clicking on an
+            // Update the selection focus only if doc is not null. When double-clicking on an
             // image file, we get a null doc here but we still want to keep _fileSelectionFocus
             // as PROJECT_MANAGER. Regardless of doc is null or not, call _activatePane
             // to trigger documentSelectionFocusChange event.
-            _fileSelectionFocus = WORKING_SET_VIEW;
+            _fileSelectionFocus = PROJECT_MANAGER;
```
Comment (humphd) [inline src/project/FileViewController.js]: Add a comment above this line:

```js
// XXXBramble: always keep focus on the Project Manager vs. Working Set View
```

File: src/project/FileViewController.js [hunk 7]
```diff
@@ -273,6 +249,5 @@ define(function (require, exports, module) {
     exports.openAndSelectDocument = openAndSelectDocument;
     exports.openFileAndAddToWorkingSet = openFileAndAddToWorkingSet;
     exports.setFileViewFocus = setFileViewFocus;
-    exports.WORKING_SET_VIEW = WORKING_SET_VIEW;
```
Comment (humphd) [inline src/project/FileViewController.js]: Restore this.

Comment (Th30) [inline src/project/FileViewController.js]: Okay, I'll take care of this tonight.

**Comment by Th30** [comment]
I'll make adjustments today for this.

**Comment by Th30** [comment]
Changes should be made, let me know if I missed something. @humphd

**Comment by flukeout** [comment]
I'll test the functionality now now @Th30 - thanks!

**Comment by flukeout** [comment]
Ok, this looks good to me - tested and it highlights the correct file when loading a project and creating a new one. If @humphd is good with the code, then let's merge it! Thanks for your work on this @Th30 👍

**Comment by humphd** [comment]
@Th30 you need to revert the changes you have to your `src/extensions/default/JavaScriptCodeHints/thirdparty/tern` and `src/extensions/default/JavaScriptCodeHints/thirdparty/acorn` submodules.  Try doing `git submodule update --init`.  Did you add these back in, by chance?  They aren't used on master anymore.

**Comment by Th30** [comment]
@humphd I've never seen those before at all:S not sure what happened.

**Comment by humphd** [comment]
You seem to have added and commit them.  Do you do `git add -a` by chance, or add with wildcards?  That's very dangerous.

Looking at your commits, it happened here:

https://github.com/mozilla/brackets/pull/688/commits/5478f40feb2d73b086d60b2ab2d61d60a7282494

I would do this:

`git revert 5478f40feb2d73b086d60b2ab2d61d60a7282494`

**Comment by Th30** [comment]
@humphd I did git add .

**Comment by humphd** [comment]
> @humphd I did git add .

And you got what you deserved!  I warned you about this in class, and it's a good lesson: don't ever `git add` with wildcard patterns, or you end up with a mess.  Always specify each file.

> The revert didn't work

What do you mean?  Look at your `git log` and it should show a new commit that undoes what you did.

**Comment by Th30** [comment]
@humphd 

It showed me an error, but after typing git status, it seems to be deleting the correct files.

**Comment by Th30** [comment]
@humphd 
error: revert is not possible because you have unmerged files.
hint: Fix them up in the work tree, and then use 'git add/rm <file>'
hint: as appropriate to mark resolution and make a commit.
fatal: revert failed

**Comment by humphd** [comment]
Git is telling you what the issue is: 

> you have unmerged files.
> hint: Fix them up in the work tree, and then use 'git add/rm '
> hint: as appropriate to mark resolution and make a commit.

It can't commit the revert because you have files that would lose your current changes.  Add them, commit, then revert.

**Comment by Th30** [comment]
@humphd 
Ok I think I got it. Hopefully...never using . ever again.

File: src/project/FileViewController.js [hunk 8]
```diff
@@ -61,24 +61,20 @@ define(function (require, exports, module) {
...
     /**
      * @private
      * @see #getFileSelectionFocus
      */
     var _fileSelectionFocus = PROJECT_MANAGER;
 
-    // Due to circular dependencies, not safe to call on() directly
-    /**
-     * Change the doc selection to the working set when ever a new file is added to the working set
-     */
```
Comment (humphd) [inline src/project/FileViewController.js]: Please add this comment back.

File: src/project/ProjectModel.js [hunk 9]
```diff
@@ -693,6 +693,7 @@ define(function (require, exports, module) {
         this.performRename();
 
         this._viewModel.moveMarker("selected", oldProjectPath, pathInProject);
+
```
Comment (humphd) [inline src/project/ProjectModel.js]: Do a `git checkout -- src/project/ProjectModel.js` to revert this change.

**Comment by humphd** [comment]
It doesn't look like you undid the changes to `ProjectModel.js` @Th30, can you do that and commit/push again?

**Comment by Th30** [comment]
@humphd hey, I tried everything but I'm just going to overwrite the file and make the applicable changes. I'll push it when I'm near my computer tonight:)

**Comment by Th30** [comment]
@humphd Hello again, if this is getting to convoluted I could do another PR to clear up some of the clutter. I got everything up and running, so let me know if there are any other changes that are required:)

**Comment by Th30** [comment]
For time saving purposes, I'm closing this PR and will reference it in the new PR.
